orttraining/orttraining/core/graph/aten_op_gradient.h

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include <vector>
+#include <string>
+#include <unordered_map>
+
+#include "core/common/common.h"
+#include "core/graph/constants.h"
+
+namespace onnxruntime {
+namespace training {
+
+typedef const char* (*GetGradientDefinitionFunc)(const char* op_name, const char* overload_name);
+
+struct PairHash {
+  template <class T1, class T2>
+  size_t operator()(const std::pair<T1, T2>& pair) const {
+    return std::hash<T1>()(pair.first) ^ std::hash<T2>()(pair.second);
+  }
+};
+
+// We try to parse more JSON contents to these structs as this is one-time effort. But for some attributes, the final
+// value and dtype is determined at runtime, e.g., when the dtype is IElemType(i) or OElemType(i), so here we still keep
+// the value in JSON format and dtype in string format. We will handle these at runtime for each node.
+struct AttributeDefinition {
+  std::string value_json;
+  std::string dtype;
+  bool is_tensor;
+};
+
+struct NodeDefinition {
+  std::string op_type;
+  std::string domain = kOnnxDomain;
+  std::vector<std::string> inputs;
+  std::vector<std::string> outputs;
+  std::unordered_map<std::string, AttributeDefinition> attributes;
+};
+
+void ParseATenOpGradientDefinition(const std::string& grad_def_json_str, std::vector<NodeDefinition>& grad_def);
+
+class ATenOpGradientDefinitionGetter {
```

Comment:
For this PR, it's ok if it only works for ATenOp. 
But we can extend this approach for defining gradients for any ops.